alstjr7437/트리/트리의-부모-찾기.py

```diff
@@ -0,0 +1,46 @@
+import sys
+sys.setrecursionlimit(10**8) #재귀 리미트
+from collections import deque
+
+n = int(input())
+
+# 트리 만들기
+tree = {i:[] for i in range(1,n+1)}
+for i in range(n-1):
+    a, b = map(int, input().split())
+    tree[a].append(b)
+    tree[b].append(a)
+# print(tree)
+
+# 부모 저장하기
+visited = [0] * (n+1)
+
+# DFS
+def dfs(a):
+    for i in tree[a]:
+        if visited[i] == 0:
+            visited[i] = a
+            # print(visited, i, a)
+            dfs(i)
+
+# BFS
+def bfs():
+    while queue:
+        a = queue.popleft()
+        for i in tree[a]:
+            if visited[i] == 0:
+                queue.append(i)
+                visited[i] = a
```

Comment:
```suggestion
            if visited[i] != 0:
                continue
            queue.append(i)
            visited[i] = a
```

이렇게 하면 depth 를 줄일 수 있습니다. 
조건문이 많아지면 장풍은 혼란을 야기합죠